boards/frdm-ke02z/board.c

```diff
@@ -0,0 +1,61 @@
+/*
+ * Copyright (C) 2018 Ishraq Ibne Ashraf
+ *
+ * This file is subject to the terms and conditions of the GNU Lesser General
+ * Public License v2.1. See the file LICENSE in the top level directory for more
+ * details.
+ */
+
+/**
+ * @ingroup     boards_frdm-ke02z
+ * @{
+ *
+ * @file
+ * @brief       Board specific implementations for the FRDM-KE02Z
+ *
+ * @author      Ishraq Ibne Ashraf <ishraq.i.ashraf@gmail.com>
+ *
+ * @}
+ */
+
+#include "board.h"
+#include "cpu.h"
+#include "periph/gpio.h"
+
+void board_init(void)
+{
+    /* Initialize the Cortex-M0+ core */
+    cpu_init();
+
+    /* System initialization */
+
+    /* SPI0 pinselect */
+    bit_set32(&(SIM->PINSEL), SIM_PINSEL_SPI0PS_SHIFT);
+    bit_set32(&(SIM->SOPT), SIM_SOPT_NMIE_SHIFT);
+    bit_set32(&(SIM->SOPT), SIM_SOPT_RSTPE_SHIFT);
+    bit_set32(&(SIM->SOPT), SIM_SOPT_SWDE_SHIFT);
+
+    /* Enable SWD and FLASH clock */
+    bit_set32(&(SIM->SCGC), SIM_SCGC_SWD_SHIFT);
+    bit_set32(&(SIM->SCGC), SIM_SCGC_FLASH_SHIFT);
+
+    /* Set BUS_CLOCK = CORE_CLOCK  */
+    bit_clear32(&(SIM->BUSDIV), SIM_BUSDIV_BUSDIV_SHIFT);
+
+    /* FEI mode by default */
+    /* Trim internal reference clock (FLL input) to 39.062kHz */
+    ICS->C3 = 0x57;
```

Comment:
With "Kinetis clock driver" you mean the MCG (Multiple Clock Driver)?

The KE02 doesn't have a MCG unit but an ISC (Internal Clock Source) unit which is quite different from the MCG unit. I can implement a driver for the ICS based on the MCG driver. But in that case I think it would better if I do that as a different issue and PR.

As of this PR for the initial support can we omit the clock driver and stick with the static default configuration for clocking?

Reply:
I'm fine with static config for now. In the future, we should rework the mcg driver into a kinetis clock driver for mcg (kinetis K), mcg lite (kl43 etc) , ics (ke0x), and scg (kl28)

Reply:
Still, the magic number 0x57 should be replaced by the vendor bit macros from mke02z4.h so it is more obvious what is being set

Reply:
It is expected that the register be loaded with a value to trim the internal reference clock which controls the FLL. The register doesn't have bit fields which control the clocking system. For now the trim value to be loaded is defined in cpu_conf_kinetis.h and the value is set as, ICS->C3 = ICS_C3_SCTRIM(ICS_TRIM_VALUE);